sources/parsers/parsers.ml

```diff
@@ -47,38 +49,36 @@ let register_parser ~lang new_parser =
     end;
   parsers := (lang, new_parser) :: !parsers
 
-let debug_no_parser_for_extension s =
-  if verbose() then
-    eprintf
-      "error: no parser registered for ext %S. Using default lang ...@." s
-
-let debug_no_parser_for_input_format s =
-  if verbose() then
-    eprintf
-      "error: no parser registered for the provided input format %S ?@." s
+let find_parser ext format =
+  try List.assoc ext !parsers
+  with Not_found ->
+    raise (ParserError ("Error: no parser registered for the provided \
+                         input format %S ?@."^format))
+
+let set_output_format f =
+  if Options.infer_output_format () then
+    match f with
+    | Options.Unknown s ->
+      Format.eprintf "Warning: The extension %s is not supported@." s
+    | fmt -> Options.set_output_format fmt
+
+let get_input_parser f =
+  set_output_format f;
+  match f with
+  | Options.Native -> find_parser ".ae" "native"
+  | Options.Smtlib2 -> find_parser ".smt2" "smtlib2"
+  | Options.Why3 -> find_parser ".why3" "why3"
+  | Options.Unknown s -> find_parser s s
 
 let get_parser lang_opt =
```

Comment:
could you rename this argument `ext_opt` or something similar, this name is actually very confusing, and all the more with the current PR which introduces a real notion of language (aka format).

Reply:
I think Github UI does not point to the right place ? I rename `f` in `fmt` since this is a format and `ext` in `ext_opt` in `find_parser`

Reply:
I meant the argument `lang_opt` of `get_parser`, ^^

Reply:
AH ok ^^, done